lib/Backend/GlobOptFields.cpp

```diff
@@ -2934,6 +2933,7 @@ GlobOpt::SetObjectTypeFromTypeSym(StackSym *typeSym, const Js::Type *type, Js::E
     {
         JsTypeValueInfo* valueInfo = JsTypeValueInfo::New(this->alloc, type, typeSet);
         valueInfo->SetSymStore(typeSym);
+        this->SetSymStoreDirect(valueInfo, typeSym);
```

Comment:
why not replace SetSymStore with SetSymStoreDirect as you've done at other places?